src/vm/methodtablebuilder.cpp

```diff
@@ -1484,6 +1484,27 @@ MethodTableBuilder::BuildMethodTableThrowing(
         }
     }
 
+#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_) 
+    // All the funtions in System.Runtime.Intrinsics.X86 are hardware intrinsics.
+    // We specially treat them here to reduce the disk footprint of mscorlib.
+    if (GetModule()->IsSystem())
```

Comment:
Actually, would it possible to more this check to `MethodTableBuilder::CheckForSystemTypes` ? We do fetch the name there already - it is better to do it just once.

Reply:
This change makes more sense, will try.

Reply:
@jkotas In `BuildMethodTableThrowing`, `CheckForSystemTypes` is called after `AllocAndInitMethodDescs();` that sets every MethodDesc to `isJitIntrinsic` or not. Would it possible to chenge these call sites order?

Reply:
Hmmm, reordering these methods would likely require more elaborate refactoring - not something to do as part of this change.

I think it is ok to leave it as is. Maybe reduce the number of cases where we fetch the name by adding `&& !bmtGenerics->HasInstantiation()` condition in addition to the IsSystem condition.

Reply:
Done.